### ISSUE TITLE
[Design] Profile: Discovery around functionality that allows users to edit contact information without abandoning their task (Part 1)

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/va.gov-team#26914]
## Background
We have an existing modal solution, but don’t think that offers the most intuitive or accessible UX.  
Let's explore how we might allow Veterans to edit contact information without leaving their current task, while considering other uses cases on VA.gov that could help make this a reusable component.

[Note from Kevin Hoffman on our slack thread about next steps](https://dsva.slack.com/archives/C909ZG2BB/p1625233376036200?thread_ts=1625082968.005300&cid=C909ZG2BB): 

> I leave it to the team to determine whether or not to address this pre- or post-launch, IMO, your findings demonstrate that the behavior this is intended to support (modifying data for a contact channel) is a secondary one to the main goal of this tool (modifying notification preferences). Therefore, post-launch makes more sense to me.

## Tasks

- [x] Identify where, if at all, we have edit-in-place functionality on VA.gov and learn what we can about how that is working for Veterans
- [x] Learn about other profile info edit-in-place use cases (check with Kristen and Kevin RE their needs)

**Comment by andaleliz** [comment]
@erikphansen and @tmitchellgcio, our team determined that edit in place is a post-MVP feature. In the interim, could we track a user (a cookie?) and if they come from notification settings and add a contact email or mobile number, show them a confirmation alert with a link to go back to notification settings? 

It'd be a band-aid to help users with the struggles we saw in usability testing until we can do the work in this ticket.

cc @ajakabcin for awareness

**Comment by erikphansen** [comment]
@andaleliz we'd have to do a research spike on that. I'd suggest we use something like a `next` URL query param. We do something similar when you try to access a page that requires auth. Example: if you go to the Profile while logged out, you get redirected to the sign-in page and the URL is `https://staging.va.gov/?next=%2Fprofile%2Fpersonal-information`.

We'd have to make sure that a `?next=returnpage` URL param would work with our deep link URLs that include something `#edit-mobile-phone-number`.

**Comment by ajakabcin** [comment]
I'm going to move this ticket to the backlog as we determined edit-in-place is a post-MVP need and I want to keep us focused on the MVP needs.

**Comment by andaleliz** [comment]
@ajakabcin in my opinion, this profile "edit in place" functionality is separate from notification preferences. We identified the need for this through the work we did on notification preferences but it would likely (hopefully) be something that could be used anywhere on the site where we are currently directing people back to profile to edit contact information in the middle of another task.

I see this more as an overall profile improvement vs explicitly tied to notification preferences.  Do you agree?

**Comment by ajakabcin** [comment]
@andaleliz yes, agreed!

**Comment by andaleliz** [comment]
Brought this up in sprint planning - we know of 2 use cases on VA.gov (letters and claims/appeals) where our code is being used in the tool to update profile info. This should be enough to go on to help us move forward.

**Comment by andaleliz** [comment]
@kristenmcconnell you were looking into "edit in place" functionality a while back, I believe for check-in. Where did you land with that?

**Comment by kristenmcconnell** [comment]
@andaleliz We are actively working on this for check-in ([Draft wireframes](https://www.sketch.com/s/e79a827e-42cf-4a82-b554-874c75b5c70e/a/QbrGr18/play)). During check-in, Veterans go through a review of all their contact (address, email, phone), insurance, next-of-kin, and emergency contact information. It is called pre-registration. We haven't usability test anything yet. I talked to Kevin Stachura about what they implemented on the "Request A Board Appeal" form. It is a modal solution that Robin Garrison implemented. I know that modals and accessibility can be tricky. I am definitely interested in collaborating on this. We are about to work on finalizing our pre-check-in design and run a usability test.

**Comment by kristenmcconnell** [comment]
@andaleliz We have the added complication that these reviews may need to take place using our LoROTA authentication method. Therefore, we will need a way to get the Veteran from the lightweight authentication to full authentication to actually make updates. I can explain more. Let me know if you want to meet.

**Comment by andaleliz** [comment]
Thanks @kristenmcconnell! This is super helpful.  A few of follow-up questions, which we can discuss in a call if it's too much to type. It is helpful for me to have the written doc but I can take notes :)

- It looks like your wireframes are using the modal solution, is that correct? 
- Going through the wireframes, I wasn't prompted to sign in at any point. I'm assuming that clicking from the text message is the LoROTA auth you're referring to? 

>Therefore, we will need a way to get the Veteran from the lightweight authentication to full authentication to actually make updates.
Has your team given this any thoughts yet?

@kevinstachura has your team usability tested the "updating contact info" modal solution Robin implemented recently?

**Comment by kristenmcconnell** [comment]
@andaleliz Right now, we are utilizing the modal solution. You are also correct that the wireframes currently don't show log in. We need to add that step. I'm not sure if that will work well with a modal. Maybe?

**Comment by andaleliz** [comment]
Thanks @kristenmcconnell.  Is clicking from the text message the LoROTA auth you're referring to?

**Comment by andaleliz** [comment]
@kristenmcconnell requested a meeting; will add notes from our convo here once that takes place tomorrow.

**Comment by andaleliz** [comment]
I put together everything I learned in[ this doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/contact-information/discovery/improving-editing-mid-task.md).  I included some basic info, given that a new team will probably pick this work up.  Will add some possible next steps next week, and then review w/ team before closing this out.
